docs/types/operators.rst

```diff
@@ -15,6 +15,14 @@ and the type of the operator's result:
    operand, use the type of the right operand,
 3. otherwise, the operation is not allowed.
 
+.. Warning::
+    The "type of the operand" is calculated through a function mobileType() that assigns, in case we're dealing with Rational Literals,
+    the largest possible type that is smaller than the size of the value. An unintended effect of this mechanism is that rational literals
+    ( that are not floats ) get converted to the largest possible integer type that just covers the size of the number.
+
+    For eg, a Rational literal 100 will get converted to uint8.
+    Changes like these can lead to overflow errors and other unintended consequences down the line during various operations.
```

Comment:
The C++ `mobileType()` function is an implementation detail so I don't think it should be mentioned in the docs.

Also, we do not have floats :) We do have fixed-point types but something like `1.23` really is just a rational literal and in our type system has the same type as integer literals. Such literals have unlimited precision so it's neither fixed- nor float-point. It's an exact representation.

I'd put that warning lower, either just below `If both are literal numbers, the operation is computed with arbitrary precision.` or at the end of the section. That paragraph already explains what a mobile type is.

And here's how I would reword it:

```rst
.. warning::
    The result of the ternary operator is never a literal, even if all of its operands are literals.
    The result type is the bigger of the mobile types of the integer operands.

    For example ``255 + (true ? 1 : 0)`` will revert due to arithmetic overflow.
    The reason is that ``(true ? 1 : 0)`` is of ``uint8`` type, which forces the addition to be performed in ``uint8`` as well,
    and 256 exceeds the range allowed for this type.

    Another consequence is that an expression like ``1.5 + 1.5`` is valid but ``1.5 + (true ? 1.5 : 2.5)`` is not.
    This is because the former is a literal expression evaluated in unlimited precision and only its final value matters.
    The latter involves a conversion of a fractional rational literal to an integer, which is currently disallowed.
```